chain/types/ethtypes/eth_transactions.go

```diff
@@ -20,16 +22,26 @@ import (
 )
 
 const (
-	EIP1559EthTxSignatureLen = 65
-	LegacyEthTxSignatureLen  = 66
-	// LegacyHomesteadEthTxSignaturePrefix defines the prefix byte used to identify the signature of a legacy Homestead Ethereum transaction.
-	LegacyHomesteadEthTxSignaturePrefix = 0x01
-	Eip1559TxType                       = 0x02
-
-	ethLegacyHomesteadTxType    = 0x00
-	ethLegacyHomesteadTxChainID = 0x00
+	EthLegacyTxType = 0x00
+	EIP1559TxType   = 0x02
 )
 
+const (
+	EthEIP1559TxSignatureLen            = 65
+	EthLegacyHomesteadTxSignatureLen    = 66
+	EthLegacyHomesteadTxSignaturePrefix = 0x01
+	EthLegacy155TxSignaturePrefix       = 0x02
```

Comment:
Since my understanding is that EIP 155 Txs are backwards compatible with homestead Txs I'm wondering if we want a separate prefix for them?  I'm pretty sure we can get away with differentiating based on the sig V value.  The question is if this would be better, worse or no different for UX.

Reply:
@ZenGround0 

I tried both approaches and honestly having a seperate type keeps the code cleaner. Having boolean flags etc makes for a lot of branches in the implementation given that the signature lengths for both tx is different and the signature verification and the signature digest construction code is also different. Even the logic to hash the tx bytes is different b/w these two tx types.

Also, note that these transaction types are never exposed to the user and the user does not even know that we're introducing an additional tx type under the hood.